src/plugins/cesium/ngv-plugin-cesium-navigation.ts

```diff
@@ -0,0 +1,320 @@
+    }
+  `;
+
+  parseViews(): void {
+    this.views = this.viewsConfig.map((v) => {
+      const positions: Cartographic[] = v.positions.map((p) =>
+        Cartographic.fromDegrees(p[0], p[1]),
+      );
+
+      const baseRect = Rectangle.fromCartographicArray(positions);
+
+      const nw = Cartographic.toCartesian(Rectangle.northwest(baseRect));
+      const ne = Cartographic.toCartesian(Rectangle.northeast(baseRect));
+      const sw = Cartographic.toCartesian(Rectangle.southwest(baseRect));
+      const se = Cartographic.toCartesian(Rectangle.southeast(baseRect));
+      const bottomPositions = [nw, ne, se, sw];
+      updateHeightForCartesianPositions(
+        bottomPositions,
+        v.elevation,
+        this.viewer.scene,
+        true,
+      );
+      const [nwt, net, set, swt] = updateHeightForCartesianPositions(
+        bottomPositions,
+        v.elevation + v.height,
+        this.viewer.scene,
+      );
+
+      const highlightEntity = this.dataSource.entities.add(
+        new Entity({
+          show: false,
+          polygon: {
+            hierarchy: bottomPositions,
+            heightReference: HeightReference.RELATIVE_TO_GROUND,
+            extrudedHeightReference: HeightReference.RELATIVE_TO_GROUND,
+            material: v.highlightColor
+              ? Color.fromCssColorString(v.highlightColor)
+              : Color.RED.withAlpha(0.6),
+            extrudedHeight: v.height,
+          },
+        }),
+      );
+
+      return {
+        top: {
+          ...calculateViewOnRectangle(swt, set, nwt, net, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        north: {
+          ...calculateViewOnRectangle(ne, nw, net, nwt, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        east: {
+          ...calculateViewOnRectangle(se, ne, set, net, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        west: {
+          ...calculateViewOnRectangle(nw, sw, nwt, swt, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        south: {
+          ...calculateViewOnRectangle(sw, se, swt, set, v.fovAngle),
+          duration: v.flyDuration,
+        },
+        title: v.title,
+        highlightEntity,
+      };
+    });
+  }
+
+  willUpdate(changedProperties: PropertyValues): void {
+    if (changedProperties.has('currentViewIndex')) {
+      this.currentView = this.views[this.currentViewIndex];
+      this.viewer.camera.flyTo(this.currentView.south);
+    }
+  }
+
+  firstUpdated(): void {
+    this.dataSourceCollection
+      .add(this.dataSource)
+      .catch((err) => console.error(err));
+    // todo make general loading?
+    const removeCallback =
+      this.viewer.scene.globe.tileLoadProgressEvent.addEventListener(
+        (tilesLoading: number) => {
+          this.tilesToLoad = tilesLoading;
+          if (tilesLoading === 0) {
+            this.parseViews();
+            this.currentViewIndex = 0;
+            removeCallback();
```

Comment:
Why do we wait for the tiles to be loaded before parsing the views?

Reply:
Also, what about "parsing" the views each time, just before using them?
That would allow to adapt to change in orientation. Let's discuss it.

Reply:
> Why do we wait for the tiles to be loaded before parsing the views?

to get the correct elevation

> Also, what about "parsing" the views each time, just before using them?
That would allow to adapt to change in orientation. Let's discuss it.

I made a parsing of a view each time but I didn't understand about orientation